dashboard/src/pages/TreeDetails/Tabs/Build/BuildTab.tsx

```diff
@@ -15,11 +17,204 @@ import { ISummary } from '@/components/Summary/Summary';
 
 import { TableFilter, TFilterObjectsKeys } from '@/types/tree/TreeDetails';
 import { ITreeDetails } from '@/pages/TreeDetails/TreeDetails';
+import { LineChart } from '@/components/LineChart';
+import BaseCard from '@/components/Cards/BaseCard';
+import { useTreeCommitHistory } from '@/api/TreeDetails';
+import { TLineChartProps } from '@/components/LineChart/LineChart';
 
 interface BuildTab {
   treeDetailsData?: ITreeDetails;
 }
 
+const StatusCard = ({
+  treeDetailsData,
+  toggleFilterBySection,
+}: {
+  toggleFilterBySection: (
+    value: string,
+    filterSection: TFilterObjectsKeys,
+  ) => void;
+  treeDetailsData?: ITreeDetails;
+}): JSX.Element => {
+  const { formatMessage } = useIntl();
+  return (
+    <BaseCard
+      title={formatMessage({ id: 'treeDetails.buildStatus' })}
+      content={
+        <StatusChartMemoized
+          type="chart"
+          pieCentralLabel={formatMessage({ id: 'treeDetails.executed' })}
+          pieCentralDescription={
+            <>
+              {(treeDetailsData?.buildsSummary.invalid ?? 0) +
+                (treeDetailsData?.buildsSummary.valid ?? 0) +
+                (treeDetailsData?.buildsSummary.null ?? 0)}
+            </>
+          }
+          onLegendClick={(value: string) => {
+            toggleFilterBySection(value, 'status');
+          }}
+          elements={[
+            {
+              value: treeDetailsData?.buildsSummary.valid ?? 0,
+              label: 'treeDetails.success',
+              color: Colors.Green,
+            },
+            {
+              value: treeDetailsData?.buildsSummary.invalid ?? 0,
+              label: 'treeDetails.failed',
+              color: Colors.Red,
+            },
+            {
+              value: treeDetailsData?.buildsSummary.null ?? 0,
+              label: 'treeDetails.unknown',
+              color: Colors.Gray,
+            },
+          ]}
```

Comment:
very nit: what about moving this to a const?

Reply:
Please memoize it since we are using `StatusChartMemoized` here